backend/app/services/ai_service_relaycore.py

```
@@ -317,28 +326,36 @@
+                    f"OpenAI API error on attempt {attempt + 1}: {e}"
+                )
                 if attempt < self.max_retries:
                     await asyncio.sleep(self.retry_delay)
                     continue
 
             except Exception as e:
                 last_error = e
-                self.logger.error(f"Unexpected error on attempt {attempt + 1}: {e}")
+                self.logger.error(
+                    f"Unexpected error on attempt {attempt + 1}: {e}"
+                )
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)